pkg/controller/vm_controller.go

```diff
@@ -338,6 +341,82 @@ func (r *VMReconciler) buildVMPod(ctx context.Context, vm *virtv1alpha1.VirtualM
 				Args:            []string{volumeMount.MountPath + "/rootfs.raw", strconv.FormatInt(volume.ContainerRootfs.Size.Value(), 10)},
 				VolumeMounts:    []corev1.VolumeMount{volumeMount},
 			})
+		case volume.PersistentVolumeClaim != nil:
+			vmPod.Spec.Volumes = append(vmPod.Spec.Volumes, corev1.Volume{
+				Name: volume.Name,
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: volume.PersistentVolumeClaim.ClaimName,
+					},
+				},
+			})
+
+			pvcKey := types.NamespacedName{
+				Namespace: vm.Namespace,
+				Name:      volume.PersistentVolumeClaim.ClaimName,
+			}
+			var pvc corev1.PersistentVolumeClaim
+			if err := r.Client.Get(ctx, pvcKey, &pvc); err != nil {
+				return nil, fmt.Errorf("get volume PVC: %s", err)
+			}
+
+			if *pvc.Spec.VolumeMode == corev1.PersistentVolumeFilesystem {
```

Comment:
Should take care of pvc.Spec.VolumeMode == nil case

Reply:
when create a PVC without spec.VolumeMode field, k8s webhook will add 'filesystem' as default

Reply:
That's OK if k8s does that, but still we should be defensive here (anywhere)